### PR TITLE
Refactor PackageController#find_last_req method

### PR DESCRIPTION
**Pull request by vpereira**
* Use guard clause instead of long condition blocks
* Extract complex ActiveRecord query to external method

### CODE REVIEW

**Review by eduardoj — APPROVED** [review]

File: src/api/app/controllers/webui/package_controller.rb [hunk 1]
```diff
@@ -831,26 +831,24 @@ def check_package_name_for_new
...
-        when: last_req.updated_at,
-        comment: last_req.comment
-      }
-    end
-    nil
+    last_req = find_last_bs_request_not_declined
+
+    return if last_req.blank?
+
+    { id: last_req.number, decliner: last_req.commenter,
```
Comment (coolo) [inline src/api/app/controllers/webui/package_controller.rb:840]: aehm, do you realize why it says decliner here? Hint: this is not about not_declined

Comment (vpereira) [reply src/api/app/controllers/webui/package_controller.rb:840]: ähhm was meinst du @coolo 👼

**Review by coolo — CHANGES_REQUESTED** [review]
The code checks the last request and only returns it if it was declined. Check where this is used:

```
    The previous request
    = link_to(request['id'], request_show_path(project, package, request['id']))
    was declined #{fuzzy_time_string(request['when'])} by
    = user_with_realname_and_icon(request['decliner'], short: true)
    with the following message:
    %pre= request['comment']
```

**Review by coolo — COMMENTED** [review]
looks more correct now

**Review by coolo — COMMENTED** [review]
Can't I just undo 'request changes?'

**Comment by danidoni** [comment]
> Can't I just undo 'request changes?'

Can't you dismiss the review?

**Comment by coolo** [comment]
thx

**Review by eduardoj — APPROVED** [review]

**Pull request merged**